apis/core/v1alpha2/core_types.go

```diff
@@ -494,8 +494,50 @@ type DataOutput struct {
 	// If no conditions is specified, it is by default to check output value not empty.
 	// +optional
 	Conditions []ConditionRequirement `json:"conditions,omitempty"`
+	// OutputToReference specifies the object used to store of intermediate data genereted by Operations
+	OutputToReference DataReference `json:"outputToRef,omitempty"`
 }
 
+// DataReference specifies the referenced object in DataOutput or DataInput
+type DataReference struct {
+	runtimev1alpha1.TypedReference `json:",inline"`
+	// Operations specify the data processing operations
+	Operations []DataOperation `json:"operations,omitempty"`
+}
+
+// DataOperation defines the specific operation for data
+type DataOperation struct {
+	// TypeName specific the type of DataOperation
+	TypeName string `json:"type"`
```

Comment:
What are the possible "type"s?

Reply:
In now implementation, only jsonPath is supported.